src/popup.js

```diff
@@ -1,19 +1,44 @@
 let quickview = document.getElementById('quickview');
 let calculation = document.getElementById('calculation');
 let liveModification = document.getElementById('liveModification');
-let percentageColumn = document.getElementById('percentageColumn');
+let percentages = document.getElementById('percentages');
+let betterTableLayout = document.getElementById('betterTableLayout');
+let percentagePosition = document.getElementById('percentagePosition');
 let version = document.getElementById('version');
 
+const setElementDisplays = () => {
+    // hide percentages position select if percentages are disabled
+    document.getElementsByClassName('percentagePosition')[0].style.display = percentages.checked ? 'block' : 'none';
+
+    // hide 6th column option if better table layout is disabled
+    if (betterTableLayout.checked && percentagePosition.length === 4) {
+        let option = document.createElement('option');
+        option.text = '6';
+        option.value = '6';
+        percentagePosition.add(option);
+    }
+    if (!betterTableLayout.checked && percentagePosition.length === 5) {
+        if (percentagePosition.value === '6')
+            percentagePosition.value = '5';
+        percentagePosition.remove(percentagePosition.length - 1);
+    }
+};
+
 const updateSettings = () => {
     let quickviewToggle = quickview.checked;
     let calculationToggle = calculation.checked;
```

Comment:
```suggestion
const updateSettings = () => {
    setElementDisplays();

    let quickviewToggle = quickview.checked;
    let calculationToggle = calculation.checked;
```

Run the element display modification before the settings are loaded and updated.